controllers/vrg_volrep.go

```diff
@@ -1476,14 +1488,14 @@ func (v *VRGInstance) deleteVR(vrNamespacedName types.NamespacedName, log logr.L
 	}
 
 	err := v.reconciler.Delete(v.ctx, cr)
-	if err == nil || k8serrors.IsNotFound(err) {
-		return nil
-	}
+	if err != nil || !k8serrors.IsNotFound(err) {
+		log.Error(err, "Failed to delete VolumeReplication resource")
 
-	log.Error(err, "Failed to delete VolumeReplication resource")
+		return fmt.Errorf("failed to delete VolumeReplication resource (%s/%s), %w",
+			vrNamespacedName.Namespace, vrNamespacedName.Name, err)
+	}
 
-	return fmt.Errorf("failed to delete VolumeReplication resource (%s/%s), %w",
-		vrNamespacedName.Namespace, vrNamespacedName.Name, err)
+	return v.ensureVRDeletedFromAPIServer(vrNamespacedName)
```

Comment:
Pass the log as well to the function, to log with relevant keys instead of using v.log as in this function.